src/particle.android.ts

```diff
@@ -31,17 +33,36 @@ export class Particle implements TNSParticleAPI {
     });
   }
 
+  public loginWithToken(token: string): void {
+    ParticleCloudSDK.getCloud().setAccessToken(token);
+    if (global["TNS_WEBPACK"]) {
+      const WorkerScript = require("nativescript-worker-loader!./particle-worker.js");
+      worker = new WorkerScript();
+      const EventWorkerScript = require("nativescript-worker-loader!./particle-event-worker.js");
```

Comment:
Do we need to do this for the other login method as well? Or better still: can we start the eventworker lazily when a subscribe call is made?

Reply:
Yes we do need that for the other login method too. Forgot to add it sorry. We could start the eventworker lazily. It needs a list of cached devices identical to the main worker though (I couldn't figure out a way to 'share' the cached devices between the workers), so maybe it should be started when calling listDevices for the first time, just before calling eventWorker.postMessage({action: "listDevices"});
What's your thoughts?

Reply:
Yep, agreed! And indeed, sharing native stuff between workers is not currently possible in NativeScript.

Reply:
I have updated this as discussed.